snapcraft/commands/snap.py

```diff
@@ -0,0 +1,77 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2015 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+snapcraft snap
+
+Creates a snap.
+
+Usage:
+  snap [options] [DIRECTORY]
+
+Options:
+  DIRECTORY             optional target directory to snap.
+  -h --help             show this help message and exit.
+
+"""
+
+import logging
+import os.path
+import subprocess
+
+import yaml
+from docopt import docopt
+
+from snapcraft import (
+    common,
+    lifecycle,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+def _snap_data_from_dir(dir):
+    # TODO: change to 'snap.yaml'
+    with open(os.path.join(dir, 'meta', 'package.yaml')) as f:
+        snap = yaml.load(f)
+
+    return {'name': snap['name'],
+            'version': snap['version'],
+            'arch': snap['architectures']}
+
+
+def main(argv=None):
+    argv = argv if argv else []
+    args = docopt(__doc__, argv=argv)
+
+    if args['DIRECTORY']:
+        # TODO: needs design
+        # TODO: overly simplistic, no meta/snap.yaml check
+        snap_dir = os.path.abspath(args['DIRECTORY'])
```

Comment:
@elopio The integration tests will come in the next PR where I will enable them all again. That `TODO` is about `snap.yaml` is outdated though.